pkg/utils/utils.go

```diff
@@ -68,6 +78,34 @@ func RewriteRegexp(operation esv1beta1.ExternalSecretRewriteRegexp, in map[strin
 	return out, nil
 }
 
+// RewriteTransform applies string transformation on each secret key name to rewrite.
+func RewriteTransform(operation esv1beta1.ExtermalSecretRewriteTransform, in map[string][]byte) (map[string][]byte, error) {
+	out := make(map[string][]byte)
+	for key, value := range in {
+		sec := &corev1.Secret{
```

Comment:
I find it a bit odd that we need to construct a v1.Secret here. I think it would be a bit cleaner if we would get the `template.FuncMap()` and construct/render the template ourselves here.
I don't think we need all the shenanigans from the `template.Execute()` func.